kroxylicious/src/main/java/io/kroxylicious/proxy/codec/KafkaResponseDecoder.java

```diff
@@ -73,16 +71,21 @@ protected Logger log() {
 
     @Override
     protected Frame decodeHeaderAndBody(ChannelHandlerContext ctx, ByteBuf in, int length) {
-        in.markReaderIndex();
-        var correlationId = in.readInt();
-        in.resetReaderIndex();
+        var wi = in.writerIndex();
+        var ri = in.readerIndex();
+        var upstreamCorrelationId = in.readInt();
+        in.readerIndex(ri);
+
+        CorrelationManager.Correlation correlation = this.correlationManager.getBrokerCorrelation(upstreamCorrelationId);
+        assert correlation != null : "Missing correlation id " + upstreamCorrelationId;
```

Comment:
Hum, hum, I got to admit I never quite got the hang of asserts. One still would run into an exception later on if the asserted condition fails and assertions are disabled (as mostly the case in prod, I suppose). If it's worthy to be enforced, why not really thrown an Exception, e.g. via `Objects::requireNonNull()`?

Reply:
I can relate about `assert`, but I think @franz1981 sneaked some in too.

```
$ grep -lir assert kroxylicious/src/
kroxylicious/src/main/java/io/kroxylicious/proxy/codec/DecodedFrame.java
kroxylicious/src/main/java/io/kroxylicious/proxy/codec/KafkaResponseDecoder.java
kroxylicious/src/main/java/io/kroxylicious/proxy/codec/OpaqueFrame.java
kroxylicious/src/main/java/io/kroxylicious/proxy/codec/KafkaRequestEncoder.java
kroxylicious/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java
kroxylicious/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyBackendHandler.java
kroxylicious/src/test/java/io/kroxylicious/proxy/codec/RequestEncoderTest.java
kroxylicious/src/test/java/io/kroxylicious/proxy/codec/ByteBufAccessorTest.java
kroxylicious/src/test/java/io/kroxylicious/proxy/codec/RequestDecoderTest.java
kroxylicious/src/test/java/io/kroxylicious/proxy/codec/AbstractCodecTest.java
kroxylicious/src/test/java/io/kroxylicious/proxy/codec/ResponseDecoderTest.java
kroxylicious/src/test/java/io/kroxylicious/proxy/ProxyTest.java
```

I'll fix this one, but let's address the rest in a separate PR.

Reply:
Perfect, thanks. Logged https://github.com/kroxylicious/kproxy/issues/45 for getting rid of them entirely.